ground/src/main/java/com/google/android/ground/ui/map/gms/features/MapsItemManager.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.ui.map.gms.features
+
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.Polygon as MapsPolygon
+import com.google.android.gms.maps.model.Polyline
+import com.google.android.ground.model.geometry.LineString
+import com.google.android.ground.model.geometry.MultiPolygon
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import com.google.android.ground.ui.map.Feature
+import com.google.maps.android.PolyUtil.*
+
+/** Manages [Feature]s displayed on the map as Maps SDK items (marker, polyline, etc). */
+class MapsItemManager(
+  private val map: GoogleMap,
+  private val pointRenderer: PointRenderer,
+  private val polygonRenderer: PolygonRenderer,
+  private val lineStringRenderer: LineStringRenderer
+) {
+  private val itemsByTag = mutableMapOf<Feature.Tag, List<Any>>()
+
+  /**
+   * Adds one or more items to the map representing the specified [Feature], replacing and existing
+   * items associated with same tag are replaced.
+   */
+  fun put(feature: Feature, visible: Boolean) =
+    with(feature) {
```

Comment:
Does `with` really add much here?

Reply:
Same comment as above. As with natural language, using a preposition can help clarity. Was `with` actually designed for use only with state mutations, or is that just a bias since it's a common pattern?

Reply:
Mainly pattern, bias I think. You've convinced me that it's fine here since I can't think of any actual problems it induces in this usage.